client/src/components/pages/wrapper-columns/config.ts

```diff
@@ -0,0 +1,41 @@
+import { IBoardTaskColumn } from '@/types';
+
+export function filterData(data: IBoardTaskColumn[], search: string) {
+  if (!search) return data;
+
+  const searchLower = search.toLowerCase();
+
+  return data
+    .map((item) => {
```

Comment:
Extracting those if condition parts into separate internal functions would be better to reduce the complexity.